File: 01/aoc202101.erl
-module(aoc202101).
-export([start/0]).

start() ->
    {ok, File} = file:open("input.txt",[read]),
    Dlist = read_int(File, [], []),
    file:close(File).

read_int(File, A) ->
    Apa = io:fread(File, "", "~d"),
    case Apa of
    eof -> 
        A;
    {ok, [N]} ->
        read_int(File, [N | A]);
    {error, What} -> 
        io:format("io:fread error: ~w~n", [What]),
        A
    end.

File: 01/aoc2021_01.py
# aoc2021_01.py

def a(depths):
    sinks = [(depths[n+1]-depths[n]) > 0 for n in range(len(depths)-1)]
    return sinks.count(True)
def b(depths):
    sinks = [(depths[n+3]-depths[n]) > 0 for n in range(len(depths)-3)]
    return sinks.count(True)

with open("input.txt") as f:
    input = f.readlines()


depthlist = [int(x) for x in input]
print(a(depthlist))
print(b(depthlist))


